src/overview.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Dart overview
+description: A short introduction to Dart
+---
+
+Dart is a client-optimized language for fast apps on any platform. Our goal is
+to offer the most productive programming language for multi-platform development,
+paired with a flexible execution runtime platform for app frameworks across
+Google.
+
+<img src="{% asset logo_lockup_dart_horizontal.png @path %}" width="300px" alt="Dart product logo">
+
+Languages are defined by their technical envelope: the many choices that are
+made during development which shape their capabilities and strengths. Dart is
+designed for a technical envelope that is particularly suited to client
+development, prioritizing both development (sub-second stateful hot reload) and
+high-quality production experiences across a wide variety of compilation targets
+(web, mobile, desktop).
+
+Dart also forms the foundation of [Flutter](https://flutter.dev): it provides
+the language and runtimes that power Flutter apps, but also supports many core
+developer tasks like formatting, analyzing, and testing code.
+
+## Dart: The Language
+
+The Dart language is type safe: it uses a combination of static type checking
+and runtime checks to ensure that a variable’s value always matches the
+variable’s static type, sometimes referred to as sound typing. Although types
+are mandatory, type annotations are optional because of type inference.
+
+Dart also offers sound null safety, meaning that values can’t be null unless you
+say they can be. With sound null safety, Dart can protect you from null
+exceptions at runtime through code analysis. Unlike many other null-safe
+languages, when Dart determines that a variable is non-nullable, that variable
+is always non-nullable: if you inspect your running code in the debugger, you’ll
+see that non-nullability is retained at runtime (hence "sound" null safety).
+
+```dart
+import 'dart:async';
+import 'dart:math' show Random;
+
+main() async {
+  print('Compute π using the Monte Carlo method.');
+  await for (final estimate in computePi().take(100)) {
+    print('π ≅ $estimate');
+  }
+}
+/// Generates a stream of increasingly accurate estimates of π.
+Stream<double> computePi({int batch: 100000}) async* {
+  var total = 0;
+  var count = 0;
+  while (true) {
+    final points = generateRandom().take(batch);
+    final inside = points.where((p) => p.isInsideUnitCircle);
+
+    total += batch;
+    count += inside.length;
+    final ratio = count / total;
+
+    // Area of a circle is A = π⋅r², therefore π = A/r².
+    // So, when given random points with x ∈ <0,1>,
+    // y ∈ <0,1>, the ratio of those inside a unit circle
+    // should approach π / 4. Therefore, the value of π
+    // should be:
+    yield ratio * 4;
+  }
+}
+Iterable<Point> generateRandom([int seed]) sync* {
```

Comment:
We should update this to null safety; in that, this needs to be `[int? seed]`. We could add a comment that explains that all other variable in this code are non-null.